### PR TITLE
[JITLink] Add x86_64::Delta8 edge kind, ELF::R_X86_64_PC8 support

### PR DESCRIPTION
**Pull request by maksfb**
Add support for ELF::R_X86_64_PC8 relocation via new x86_64::Delta8 edge kind.

### CODE REVIEW

File: llvm/include/llvm/ExecutionEngine/JITLink/x86_64.h [hunk 1]
```diff
@@ -473,6 +486,15 @@ inline Error applyFixup(LinkGraph &G, Block &B, const Edge &E,
     break;
   }
 
+  case Delta8: {
+    int64_t Value = E.getTarget().getAddress() - FixupAddress + E.getAddend();
+    if (LLVM_LIKELY(isInt<8>(Value)))
+      *(little32_t *)FixupPtr = Value;
```
Comment (lhames) [inline llvm/include/llvm/ExecutionEngine/JITLink/x86_64.h:492]: I think `little32_t` will lead to overflow here. You can just write to `FixupPtr` without casting (it's already a `char*`), or make it explicit:
```c++
*(char*)FixupPtr = Value;
```

**Review by lhames — CHANGES_REQUESTED** [review]

File: llvm/include/llvm/ExecutionEngine/JITLink/x86_64.h [hunk 2]
```diff
@@ -95,6 +95,19 @@ enum EdgeKind_x86_64 : Edge::Kind {
   ///
   Delta32,
 
+  /// An 8-bit delta.
+  ///
+  /// Delta from the fixup to the target.
+  ///
+  /// Fixup expression:
+  ///   Fixup <- Target - Fixup + Addend : int64
```
Comment (lhames) [inline llvm/include/llvm/ExecutionEngine/JITLink/x86_64.h:103]: I just noticed that this comment should have `: int8` too.

**Review by lhames — APPROVED** [review]
Otherwise LGTM. Thanks very much @maksfb!

File: llvm/include/llvm/ExecutionEngine/JITLink/x86_64.h (hunk 2 above)
Comment (maksfb) [reply llvm/include/llvm/ExecutionEngine/JITLink/x86_64.h]: Makes sense. Fixed the comment for `Delta32` above as well.

Comment (lhames) [reply llvm/include/llvm/ExecutionEngine/JITLink/x86_64.h]: Oh nice catch, thank you!

**Pull request merged**